2023-02-26-binplz.md

```diff
@@ -0,0 +1,116 @@
+------------------------------------------------------
+title: binplz: easily curl static binaries
+summary: Introducing binplz, a service that serves curl-able statically-linked binaries from Nixpkgs
+tags: nixos
+draft: false
+------------------------------------------------------
+
+[Nixery](https://nixery.dev/) is an interesting service. It dynamically serves
+Docker images created from arbitrary packages from Nixpkgs.
+
+I thought that a similar type of service could be created that serves
+statically-linked binaries from Nixpkgs.  This would be another way for users
+on any Linux distro to benefit from all the software packaged in Nixpkgs.  I
+explained this to my friends [Jonas Carpay](https://jonascarpay.com/)
+([@jonascarpay](https://github.com/jonascarpay)) and Hideaki Kawai
+([@kayhide](https://github.com/kayhide)), and we came up with the name
+`binplz.dev`.  I was thinking that the service would work like the following.
+
+Imagine you're on your Linux machine, and you realize you need the `tree` command.
+Normally you could install it with your package manager, but maybe your package manager
+is broken.  (Or maybe you don't want `tree`, but you want some package that isn't
+available on your distro.)
+
+You should be able to download a statically-linked `tree` binary from `binplz.dev`:
+
+```console
+$ curl binplz.dev/tree > tree
+$ chmod +x ./tree
+```
+
+You can then use this tree binary like you'd expect:
+
+```console
+$ ./tree --version
+tree v1.8.0 (c) 1996 - 2018 by Steve Baker, ...
+```
+
+You can confirm that this tree binary is statically linked:
+
+```console
+$ file ./tree
+./tree: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, not stripped
+```
+
+The "statically linked" part of the output of file signifies that this binary
+is statically linked.
+
```

Comment:
```suggestion
```

I think you can safely remove it, it follows directly from the previous lines

Reply:
Good idea, removed in 7698efe.